huntsman/utils/config.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jan  2 12:04:11 2020
+
+@author: danjampro
+
+Code to provide a config server using pyro.
+"""
+import os, time
+import Pyro4
+from huntsman.utils import load_config, get_own_ip
+
+#==============================================================================
+
+@Pyro4.expose
+class ConfigServer():
+    '''
+    Class representing the config server.
+    '''
+    def __init__(self, config_file=None, **kwargs):
+        '''
+        
+        '''
+        if config_file is None:
+            config_file = os.path.join(os.environ['HUNTSMAN_POCS'],
+                                       'conf_files', 'camera_info_local.yaml')
+            
+        #Read the config file
+        self.config = load_config(config_files=[config_file], **kwargs)
+        
+        
+    def get_config(self, key=None):
+        '''
+        Retrieve the config file.
+        '''
+        if key is None:
+            return self.config
+        return self.config[key]
+    
+#==============================================================================
+
+def locate_name_server(wait=None, logger=None):
+    '''
+    Locate and return the name server, waiting if necessary.
+    
+    Parameters
+    ----------
+    wait (float or None) [seconds]:
+        If not None, attempt to locate the NS at this frequency. 
+
+    Returns
+    -------
+    Pyro name server.        
+    '''
+    if wait is None:
+        return Pyro4.locateNS()
+    
+    while True:
+        try:
+            return Pyro4.locateNS()
+        except Pyro4.errors.PyroError:
+            msg = 'Unable to locate name server. Waiting...'
+            if logger is None:
+                print(msg)
+            else:
+                logger.debug(msg)
+            time.sleep(wait)
+        
+        
+def start_config_server(host=None, port=6563, name='config_server',
```

Comment:
should this and the next function be a method of the `ConfigServer` class?

Reply:
Nice idea - yes, probably. The only thing that stopped me doing this is the need to register a ConfigServer object with the NS. Perhaps self could be used, but this is probably simpler for now

Reply:
I see. Makes sense.